rest/rest-resources/src/main/java/org/apache/brooklyn/rest/util/MultiSessionAttributeAdapter.java

```diff
@@ -638,12 +638,22 @@ public MultiSessionAttributeAdapter resetExpiration() {
         Handler[] hh = getSessionHandlers();
         if (hh!=null) {
             for (Handler h: hh) {
-                Session ss = ((SessionHandler)h).getSession(getId());
-                if (ss!=null) {
+                Session ss = getSessionSafely(h, getId());
+                if (ss != null) {
                     ss.setMaxInactiveInterval(maxInativeInterval);
                 }
             }
         }
         return this;
     }
+
+    private static Session getSessionSafely(Handler h, String id) {
+        if (((SessionHandler)h).getSessionCache()==null) {
```

Comment:
Check if Handler is an instance of SessionHandler first, log debug/warning message "{h} is not a session handler" and return null.

Reply:
+1 worth adding now that it is refactored